src/Cutscenes.ts

```diff
@@ -1637,6 +1637,146 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
+            },
+            1,
+            Infinity);
+
+        /*Make attacker disappear*/
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                attacker.hidden = !attacker.hidden;
+            },
+            (dt + 10) / 2);
+        /*Make attacker reappear*/
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                attacker.hidden = !attacker.hidden;
+            },
+            dt * 2);
+        /*Move attacker to original position*/
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                xvel *= -1;
+            },
+            dt);
+
+        this.EightBitter.TimeHandler.addEvent(this.EightBitter.TimeHandler.cancelEvent, dt * 2 - 1, movement);
+
+        /*const animateExplosion: (x: number, y: number) => void = (x: number, y: number): void => {
+            const explosion: IThing = this.EightBitter.ObjectMaker.make("ExplosionSmall");
+            this.EightBitter.things.add(explosion, x, y);
+            /*this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    this.EightBitter.physics.killNormal(explosion);
+                },
+                dt + 10);
+            console.log(x);
+            console.log(y);
+        };*/
+
+        const explosions: IThing[] = [
+            this.EightBitter.ObjectMaker.make("ExplosionSmall"),
+            this.EightBitter.ObjectMaker.make("ExplosionSmall"),
+            this.EightBitter.ObjectMaker.make("ExplosionSmall")
+        ];
+        let startX: number[] = [];
+        let startY: number[] = [];
+        console.log(menu.right);
+        if (direction === -1) {
+            startX[0] = menu.right - defender.width / 2 * this.EightBitter.unitsize;
+            startY[0] = menu.top;
+        } else {
+            console.log(direction);
+            startX[0] = menu.left + (defender.width + 32) * this.EightBitter.unitsize;
+            startY[0] = menu.bottom - (defender.height + 16) * this.EightBitter.unitsize;
+            startX[1] = startX[0] + 6 * this.EightBitter.unitsize;
+            startY[1] = startY[0] - 6 * this.EightBitter.unitsize;
+            startX[2] = startX[1] + 6 * this.EightBitter.unitsize;
+            startY[2] = startY[1] - 8 * this.EightBitter.unitsize;
+            console.log(startX);
+            console.log(startY);
+        }
+
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                this.EightBitter.things.add(explosions[0], startX[0], startY[0]);
+            },
+            dt);
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                this.EightBitter.physics.killNormal(explosions[0]);
+            },
+            dt + 4);
+
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                this.EightBitter.things.add(explosions[1], startX[1], startY[1]);
+            },
+            dt + 4);
```

Comment:
If two events have the same delay (`dt + 4` and `dt +4`), it would be easier to use the same function for them.